generators/craft/index.js

```diff
@@ -0,0 +1,190 @@
+        this.templatePath('craft/config/general.php'),
+        this.destinationPath('craft/config/general.php')
+      );
+
+      // Database
+      this.fs.copy(
+        this.templatePath('craft/config/db.php'),
+        this.destinationPath('craft/config/db.php')
+      );
+
+      // Asset Rev
+      if (this.options.craftPlugins.indexOf('assetrev') > -1) {
+        this.fs.copy(
+          this.templatePath('craft/config/assetrev.php'),
+          this.destinationPath('craft/config/assetrev.php')
+        );
+      }
+
+      // Imager
+      if (this.options.craftPlugins.indexOf('imager') > -1) {
+        this.fs.copyTpl(
+          this.templatePath('craft/config/imager.php'),
+          this.destinationPath('craft/config/imager.php'), {
+            projectName: this.options.projectName
+          }
+        );
+      }
+    },
+
+    templates: function() {
+      this.fs.copy(
+        this.templatePath('craft/templates/_layout.html'),
+        this.destinationPath('craft/templates/_layout.html')
+      );
+      this.fs.copy(
+        this.templatePath('craft/templates/index.html'),
+        this.destinationPath('craft/templates/index.html')
+      );
+      this.fs.copy(
+        this.templatePath('craft/templates/404.html'),
+        this.destinationPath('craft/templates/404.html')
+      );
+      this.fs.copy(
+        this.templatePath('craft/templates/_partials'),
+        this.destinationPath('craft/templates/_partials')
+      );
+    },
+
+    downloadPlugins: function() {
+      progress.start('Installing Craft Plugins');
+      var installationPromises = [];
+      var self = this;
+      this.options.craftPlugins.forEach(function(option) {
+        var plugin = plugins[option];
+        if (plugin.src.indexOf('http') == -1) return;
+        var downloadPromise = download(plugin.src, self.destinationPath('craft/plugins/downloads'), {
+          extract: true
+        })
+        installationPromises.push(downloadPromise);
+      });
+
+      return Promise.all(installationPromises).then(function() {
+        progress.stop();
+      });
+    },
+
+    movePlugins: function() {
+      var self = this;
+      this.options.craftPlugins.forEach(function(option) {
+        var plugin = plugins[option];
+        if (plugin.src.indexOf('http') == -1) return;
+        var pluginDirPath = self.destinationPath('craft/plugins/downloads/' + plugin.githubName + '-master/' + option);
+        if (!fs.existsSync(pluginDirPath)) return;
+        fs.renameSync(pluginDirPath, self.destinationPath('craft/plugins/' + option));
+      });
+      del.sync([self.destinationPath('craft/plugins/downloads/')]);
+    },
+
+    permissions: function() {
+      console.log(chalk.green('> Setting global permissions to 755 / 644'));
+      child_process.execSync('chmod -R 755 *');
+      child_process.execSync('find . -type f -exec chmod 644 {} \\;');
+      console.log(chalk.green('> Setting permissions on craft/app to 775 / 644'));
```

Comment:
Oh, I missed something critical! He's using 644 (globally) and 664 for the individual dirs. That gives group write permissions, which is a good idea. Disregard my previous comment.